src/Symfony/Component/Workflow/Registry.php

```diff
@@ -23,19 +24,34 @@ class Registry
 
     /**
      * @param Workflow $workflow
-     * @param string   $className
+     * @param string|SupportStrategyInterface $supportStrategy
      */
-    public function add(Workflow $workflow, $className)
+    public function add(Workflow $workflow, $supportStrategy)
     {
-        $this->workflows[] = array($workflow, $className);
+        if (!$supportStrategy instanceof SupportStrategyInterface) {
```

Comment:
No BC break anymore after doing suggested changes thanks to @iltar Though it seems triggering the deprecated error leads to CI integrations to fail... Any ideas how to fix this?